ci/scripts/run-codebuild/README.md

```diff
@@ -0,0 +1,39 @@
+# Run Codebuild
+
+This script executes a CodeBuild build, allowing override of the source (and secondary source) versions, as well as the environment variables of the build. It will execute the build and then retrieve the logs from CloudWatch and print them to stdout.
+
+This script expects to be run with an environment variable `PATH_TO_CONFIG` set which specifies a path relative (or an absolute path) to the `run-codebuild.js` script. This config defines which codebuild project to run and what params to pass.
+
+## Environment Variables
+
+variable | description
+---|---
+PATH\_TO\_CONFIG | A path (absolute, or relative to the run-codebuild.js script) to the JSON config file
+
+## Config
+
+Simple example config
+```
+{
+    "projectName": "codebuild-e2e-spike-test-12",
+    "sourceVersion": "123456",
+    "secondarySourcesVersions": {
+      "pay_frontend": "654321",
+      "pay_foobar": "987654"
+      
+    },
+    "environmentVariables": {
+        "PAY_FRONTEND_RELEASE": "1734-release",
+        "END_TO_END_TEST_SUITE": "products"
+    }
+}
+```
+
+The fields in this are:
+
+field | description
+---|----
+projectName | The AWS CodeBuild project name to run
+sourceVersion | The source version for the primary source of the CodeBuild project
+secondarySourcesVersions | A map with keys for the secondary source names and values as the versions to use for those sources
```

Comment:
I'm not sure why `secondarySourcesVersions` is needed. Is it related to `sourceVersion`?

Reply:
Is `sourceVersion` the source version of the CodeBuild project?
From the example config it looks to me like `secondarySourcesVersions` is referencing some frontend version but there's already `PAY_FRONTEND_RELEASE` which leaves me confused.

Reply:
Codebuild builds take multiple sources.

It's easier to conceptualise it as github repos rather than files on s3 so lets think of it in that regard for a moment.

A build could have say pay-frontend as it's source (also referred to as the primary source in some of the aws docs). But also have pay-ci and pay-infra as secondary sources.

When you execute the build you specify the sourceVersion as the github ref for the version you want to pull of pay-frontend. You specify refs for both pay-ci and pay-infra in secondarySourcesVersions

In the case of the end to end tests we only have a single source, and the source is from S3, this means we specify the version ID of the file in S3 in these (which was obtained by the upload

Reply:
> When you execute the build you specify the sourceVersion as the github ref for the version you want to pull of pay-frontend.

But I'm just seeing
```
"sourceVersion": "123456"
```
How do I infer it's a version for pay-frontend in this case?

Reply:
> In the case of the end to end tests we only have a single source, and the source is from S3, this means we specify the version ID of the file in S3

Presumably this source is the pay-ci repo?

Reply:
The source is the zip of the pay-ci file on s3. So the sourceversion is the version id of that file.

The run-codebuild task is generic to run any codebuild, so you have to know what the codebuild you are executing has configured as it's source to know what the sourceVersion refers to. In the case of secondary sources they are explicitly named.

In the tests the 123456 is just some fake data to test that what gets sent to codebuild is the same as what was specified by the request.

Reply:
> The run-codebuild task is generic to run any codebuild, so you have to know what the codebuild you are executing has configured as it's source to know what the sourceVersion refers to.

I understand. I think it's potentially confusing but it's one of things where I'll have to see its use within concourse to get a better handle of it. I'm not suggesting any code changes here.